core/src/replay_stage.rs

```diff
@@ -149,6 +149,62 @@ struct SkippedSlotsInfo {
     last_skipped_slot: u64,
 }
 
+struct PartitionInfo {
+    partition_exists: bool,
```

Comment:
nit: seems these two could be combined as `partition_exists` is `true` iff `partition_start_time` is `Some`